src/main/flight/autopilot.c

```diff
@@ -22,38 +22,141 @@
 
 #include "platform.h"
 #include "build/debug.h"
+#include "common/axis.h"
+#include "common/filter.h"
 #include "common/maths.h"
+#include "common/vector.h"
+#include "fc/rc.h"
+#include "fc/runtime_config.h"
 
 #include "flight/imu.h"
-#include "flight/pid.h"
 #include "flight/position.h"
 #include "rx/rx.h"
+#include "sensors/gyro.h"
 
 #include "autopilot.h"
 
 #define ALTITUDE_P_SCALE  0.01f
 #define ALTITUDE_I_SCALE  0.003f
 #define ALTITUDE_D_SCALE  0.01f
 #define ALTITUDE_F_SCALE  0.01f
+#define POSITION_P_SCALE  0.0012f
+#define POSITION_I_SCALE  0.0001f
+#define POSITION_D_SCALE  0.0015f
+#define POSITION_A_SCALE  0.0008f
+#define UPSAMPLING_CUTOFF_HZ 5.0f
 
 static pidCoefficient_t altitudePidCoeffs;
+static pidCoefficient_t positionPidCoeffs;
+
 static float altitudeI = 0.0f;
 static float throttleOut = 0.0f;
 
+typedef struct {
```

Comment:
Sidenote: The name "earthFrame" doesn't really make sense for this struct but since this is encapsulated in autopilot.c and not exposed to the outside, I don't really care. Can be renamed later.

Just give the struct a tag name
```suggestion
typedef struct earthFrame_s {
```

Reply:
OK, done.  
I gave this struct a name using 'earthFrame` since it carries only 'earth frame' related values, eg East-West and North parameters only.  Happy to change to a different name if you can think of a better one.